File: fix_app.js
const fs = require('fs');
const path = require('path');

const appPath = path.join(__dirname, 'frontend', 'src', 'App.jsx');

console.log("Reading App.jsx...");
let content = fs.readFileSync(appPath, 'utf8');

// Normalize line endings to LF for easier manipulation
content = content.replace(/\r\n/g, '\n').replace(/\r/g, '\n');

// 1. Extract schedulePreview block
const schedRegex = /(  const \[schedulePreview[\s\S]*?schedulePreview\.manualOverride\]\);\n)/;
const matchSched = content.match(schedRegex);

if (matchSched) {
    const schedBlock = matchSched[1];
    content = content.replace(schedBlock, "");
    
    // 2. Find newTaskData block
    const newTaskRegex = /(  const \[newTaskData, setNewTaskData\] = useState\(\{[\s\S]*?risk: 'medium'\n  \}\);)/;
    const matchNewTask = content.match(newTaskRegex);
    
    if (matchNewTask) {
        const newTaskBlock = matchNewTask[1];
        
        // 3. Insert schedulePreview AFTER newTaskData
        content = content.replace(newTaskBlock, newTaskBlock + "\n\n" + schedBlock);
        
        fs.writeFileSync(appPath, content, 'utf8');
        console.log("Successfully fixed the hook order in App.jsx! Vite should hot-reload and the crash will disappear.");
    } else {
        console.log("Error: Could not find the newTaskData block. It may have been modified.");
    }
} else {
    console.log("Error: Could not find the schedulePreview block. The patch may not have applied correctly.");
}

File: fix_app.py
import sys
import re

app_jsx_path = r'c:\Users\saura\ParseOps\frontend\src\App.jsx'

with open(app_jsx_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Pattern to capture the entire schedulePreview block (useState + useEffect)
pattern_schedule = r"(  const \[schedulePreview.*?\[newTaskData\.assignees.*?schedulePreview\.manualOverride\]\);\s*)"

match_schedule = re.search(pattern_schedule, content, re.DOTALL)
if match_schedule:
    extracted_block = match_schedule.group(1)
    
    # Remove it from content
    content = content.replace(extracted_block, "")
    
    # Now find newTaskData state initialization and insert AFTER it
    pattern_new_task = r"(  const \[newTaskData, setNewTaskData\] = useState\(\{[\s\S]*?risk: 'medium'[\s\S]*?\}\);)"
    match_new_task = re.search(pattern_new_task, content)
    
    if match_new_task:
        new_task_block = match_new_task.group(1)
        # Replace the new_task_block with itself + extracted_block
        content = content.replace(new_task_block, new_task_block + "\n\n" + extracted_block)
        
        with open(app_jsx_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print("Successfully fixed the initialization order in App.jsx! The app should now work without crashing.")
    else:
        print("Failed to find newTaskData block. Please check the file contents.")
else:
    print("Failed to find schedulePreview block. Please check the file contents.")
